sdk/com.ibm.sbt.web/src/main/webapp/js/sdk/_bridges/dojo-amd/RequestTransport.js

```diff
@@ -55,7 +55,19 @@ define([ "dojo/_base/declare", "dojo/_base/lang", "dojo/request", "../util" ], f
          * 
          */
         request : function(url,options) {
-            return request(url, options);
+        	var promise = new Promise();
+            promise.response = new Promise();
+            request(url, args).response.then(
+                function(response){
```

Comment:
Dojo/request has the concept of different provider support which dojo/xhr doesnt have. I think that the return type from Transport.js and RequestTransport.js should be same, so that the consumers consuming the APIs can seamlessly switch between the two APIs. Please correct me if I am inaccurate here.

Right now the returned promise from RequestTransport and Transport do not match. In transport, we construct a special response object (for a use case) and is platform agnostic. This is absent in the implementation of RequestTransport. 

Also in the line promise.response.fullfilled(response), we are passing a promise whereas in the transport implemenation we are not passing a promise.

Also want to understand, is this API being consumed in the code anywhere ? We might need to change there as well.